src/parser.rs

```diff
@@ -415,20 +415,19 @@ impl Library {
                                         "name" => field_name = attr.value.clone(),
                                         _ => {},
                                     }
+                                    break;
                                 }
 
                                 u = Union {
                                     name: format!(
-                                        "{}{}_{}{}",
-                                        parent_name_prefix.unwrap_or(""),
-                                        record_name,
+                                        "{}_{}{}",
+                                        parent_name_prefix.unwrap_or(record_name),
                                         field_name,
                                         union_count
                                     ),
                                     c_type: Some(format!(
-                                        "{}{}_{}{}",
-                                        parent_ctype_prefix.unwrap_or(""),
-                                        c_type,
+                                        "{}_{}{}",
+                                        parent_ctype_prefix.unwrap_or(c_type),
```

Comment:
I don't see that being a problem as since it uses the parent or c_type, it already has one part of a unique name, adding the field and count makes it definitely unique. I reckon the chances of finding something that nests structs/unions with same field names deep enough for it to become an issue are nil.